src/api/EscargotPublic.cpp

```diff
@@ -1951,6 +1962,46 @@ unsigned ArrayBufferViewRef::bytelength()
     return toImpl(this)->bytelength();
 }
 
+Int8ArrayObjectRef* Int8ArrayObjectRef::create(ExecutionStateRef* state)
+{
+    return toRef(new Int8ArrayObject(*toImpl(state)));
+}
+
+Uint8ArrayObjectRef* Uint8ArrayObjectRef::create(ExecutionStateRef* state)
+{
+    return toRef(new Uint8ArrayObject(*toImpl(state)));
+}
+
+Int16ArrayObjectRef* Int16ArrayObjectRef::create(ExecutionStateRef* state)
+{
+    return toRef(new Int16ArrayObject(*toImpl(state)));
+}
+
+Uint16ArrayObjectRef* Uint16ArrayObjectRef::create(ExecutionStateRef* state)
+{
+    return toRef(new Uint16ArrayObject(*toImpl(state)));
+}
+
+Uint32ArrayObjectRef* Uint32ArrayObjectRef::create(ExecutionStateRef* state)
+{
+    return toRef(new Uint32ArrayObject(*toImpl(state)));
+}
+
+Int32ArrayObjectRef* Int32ArrayObjectRef::create(ExecutionStateRef* state)
+{
+    return toRef(new Int32ArrayObject(*toImpl(state)));
+}
+
+Float32ArrayObjectRef* Float32ArrayObjectRef::create(ExecutionStateRef* state)
+{
+    return toRef(new Float32ArrayObject(*toImpl(state)));
+}
+
+Float64ArrayObjectRef* Float64ArrayObjectRef::create(ExecutionStateRef* state)
+{
```

Comment:
Check pointer-type parameter / check before deferencing
ASSERT(state != nullptr)
--------------to here